neo4j/_async/work/result.py

```diff
@@ -145,6 +146,11 @@ async def on_failed_attach(metadata):
     def _pull(self):
         def on_records(records):
             if not self._discarding:
+                records = (
+                    record.raw_data
+                    if isinstance(record, BrokenHydrationObject) else record
+                    for record in records
+                )
```

Comment:
Why do we need this special behaviour?

Reply:
That is because the `RECORD` message has one field which is the list of the record's values. That list will if any of its elements is a `BrokenHydrationObject` be reduced to a `BrokenHydrationObject` itself
https://github.com/neo4j/neo4j-python-driver/pull/759/files#diff-2a37374b359d1060498f6ecb133093a8ae46b21d00231fe42d8160ffc363e8edR68
Therefore, I need to unpack it for the `Record` class to be able to access the values.